packages/xo-server-netbox/src/index.js

```diff
@@ -233,6 +239,107 @@ class Netbox {
 
     log.info(`Synchronizing ${xoPools.length} pools with Netbox`, { pools: xoPools })
 
+    // Tenants -----------------------------------------------------------------
+
+    let nbTenants
+    if (this.#syncUsers) {
+      log.info('Synchronizing users')
+
+      const createNbTenant = xoUser => {
+        const name = xoUser.email.slice(0, NAME_MAX_LENGTH)
+        return {
+          custom_fields: { uuid: xoUser.id },
+          name,
+          slug: slugify(name),
+          description: 'XO user',
+        }
+      }
+
+      const xoUsers = await this.#xo.getAllUsers()
+
+      nbTenants = keyBy(await this.#request('/tenancy/tenants/'), 'custom_fields.uuid')
+      delete nbTenants.null // Ignore tenants that don't have a UUID
+
+      const nbTenantsToCheck = { ...nbTenants }
+
+      const tenantsToUpdate = []
+      const tenantsToCreate = []
+      for (const xoUser of xoUsers) {
+        const nbTenant = nbTenants[xoUser.id]
+        delete nbTenantsToCheck[xoUser.id]
+
+        const updatedTenant = createNbTenant(xoUser)
+
+        if (nbTenant !== undefined) {
+          // Tenant was found in Netbox: update it
+          const patch = diff(updatedTenant, nbTenant)
+          if (patch !== undefined) {
+            tenantsToUpdate.push(patch)
+          }
+        } else {
+          // Tenant wasn't found: create it
+          tenantsToCreate.push(updatedTenant)
```

Comment:
Why are you creating tenants for XO users that are not attached to a Netbox VM? I believe users would prefer to see in their Netbox only those users who are related to the pools they synchronize.

Reply:
Because the feature is to synchronize users (ie all the users). And a use case for that could be to assign a user to another type of object manually from Netbox so I think it's still interesting to sync them all, especially since it's an opt-in feature.